### ISSUE TITLE
1.12.0+ Required Query Params parsing fails

### ISSUE THREAD

**Issue by FloppyNotFound** [deepmap/oapi-codegen#856]
Starting from v1.12.0, generation fails, if the YAML includes required query params.

Example YAML:
```
openapi: "3.0.0"

info:
  version: 1.0.0
  title: Lorem Ipsum

paths:
  /lorem-ipsum:
    get:
      summary: Lorem Ipsum
      parameters:
        - in: query
          name: amount
          required: true
          schema:
            type: number
      responses:
        200:
          description: Success
```

Example calls:
```
./oapi-codegen -generate types -o api/api.types.gen.go -package api my.api.yaml
./oapi-codegen -generate gin,spec -o api/api.gen.go -package api my.api.yaml
```

Error message:
`api\api.gen.go:150:7: siw.ErrorHandler(c, fmt.Errorf("Query argument amount is required, but not found: %s", err), http.StatusBadRequest) (no value) used as value`

Environment:
$ go version
go version go1.18.6 windows/amd64

**Comment by schoenenberg** [comment]
Had the same issue. Upgrading to newest version fixed it for me (see also https://github.com/deepmap/oapi-codegen/pull/864).

**Comment by FloppyNotFound** [comment]
Can confirm that this issue seems to be fixed in version v1.12.3
